packages/components/src/form-token-field/suggestions-list.tsx

```diff
@@ -103,10 +105,18 @@ export function SuggestionsList< T extends string | { value: string } >( {
 		>
 			{ suggestions.map( ( suggestion, index ) => {
 				const matchText = computeSuggestionMatch( suggestion );
+				const isSelected = index === selectedIndex;
+				const isDisabled =
+					typeof suggestion === 'object' && suggestion?.disabled;
```

Comment:
Do we need the `object` type check? Calling `stringVal?.disabled` should be just fine for a string, it will just return `undefined`.

Reply:
`stringVal?.disabled`, considering `stringVal` is a String, generates a TS error:

```
Property 'disabled' does not exist on type 'string | { value: string; disabled?: boolean | undefined; }'.
  Property 'disabled' does not exist on type 'string'.ts.
```

So we need to check the type before trying to access to the `disabled` property